File: server/src/test/java/com/adobe/testing/s3mock/dto/ListPartsResultTest.java
/*
 *  Copyright 2017-2021 Adobe.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *          http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package com.adobe.testing.s3mock.dto;

import static com.adobe.testing.s3mock.dto.DtoTestUtil.serializeAndAssert;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInfo;

class ListPartsResultTest {

  @Test
  void testSerialization(TestInfo testInfo) throws IOException {
    ListPartsResult iut =
        new ListPartsResult("bucketName", "fileName", "uploadId",
            createParts(2));

    serializeAndAssert(iut, testInfo);
  }

  private List<Part> createParts(int count) {
    List<Part> parts = new ArrayList<>();
    for (int i = 1; i <= count; i++) {
      Part part = new Part();
      part.setPartNumber(i);
      part.setETag("etag" + i);
      part.setLastModified(new Date(1514477008120L));
      part.setSize(10L + i);
      parts.add(part);
    }
    return parts;
  }
}

File: server/src/test/kotlin/com/adobe/testing/s3mock/dto/ListPartsResultTest.kt
/*
 *  Copyright 2017-2025 Adobe.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *          http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package com.adobe.testing.s3mock.dto

import com.adobe.testing.s3mock.DtoTestUtil.serializeAndAssertXML
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInfo
import java.util.Date

internal class ListPartsResultTest {
  @Test
  fun testSerialization(testInfo: TestInfo) {
    val iut =
      ListPartsResult(
          "bucketName",
          ChecksumAlgorithm.CRC32,
          ChecksumType.COMPOSITE,
          Initiator("id", "displayName"),
          false,
          "fileName",
          1000,
          100,
          Owner("displayName"),
          createParts(),
          5,
          StorageClass.STANDARD,
          "uploadId",
      )
    assertThat(iut).isNotNull()
    serializeAndAssertXML(iut, testInfo)
  }

  private fun createParts(count: Int = 2): List<Part> {
    return (1..count).map { Part(it, "etag$it", Date(1514477008120L), 10L + it) }
  }
}
